mm/2s2h/Rando/ActorBehavior/ObjGrass.cpp

```diff
@@ -1041,80 +1028,35 @@ void SpawnGrassDrop(Vec3f pos, RandoCheckId randoCheckId) {
-//         return;
-//     }
-//
-//     RandoItemId randoItemId = Rando::ConvertItem(RANDO_SAVE_CHECKS[OBJGRASS_RC].randoItemId,
-//     (RandoCheckId)OBJGRASS_RC); RandoItemType randoItemType = Rando::StaticData::Items[randoItemId].randoItemType;
-//
-//     switch (randoItemType) {
-//         case RITYPE_BOSS_KEY:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotBossKeyDL);
-//             break;
-//         case RITYPE_HEALTH:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotHeartDL);
-//             break;
-//         case RITYPE_LESSER:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotMinorDL);
-//             break;
-//         case RITYPE_MAJOR:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotMajorDL);
-//             break;
-//         case RITYPE_MASK:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotMaskDL);
-//             break;
-//         case RITYPE_SKULLTULA_TOKEN:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotTokenDL);
-//             break;
-//         case RITYPE_SMALL_KEY:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotSmallKeyDL);
-//             break;
-//         case RITYPE_STRAY_FAIRY:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotFairyDL);
-//             break;
-//         default:
-//             Gfx_DrawDListOpa(play, (Gfx*)gPotStandardDL);
-//             break;
-//     }
-// }
-
 void Rando::ActorBehavior::InitObjGrassBehavior() {
-    /*COND_VB_SHOULD(VB_GRASS_DRAW_BE_OVERRIDDEN, IS_RANDO, {
-        Actor* actor = va_arg(args, Actor*);
-        if (OBJGRASS_RC != RC_UNKNOWN) {
-            *should = false;
-            actor->draw = ObjGrass_RandoDraw;
-        }
-    });*/
-
     COND_VB_SHOULD(VB_GRASS_DROP_COLLECTIBLE, IS_RANDO, {
         auto actorId = static_cast<ActorId>(va_arg(args, int32_t));
         ObjGrassElement* grassElemActor;
         EnKusa* kusaActor;
         ObjGrassCarry* grassCarryActor;
-        Vec3f pos = gZeroVec3f;
+        Vec3f GrassIdPos = gZeroVec3f;
+        Vec3f collectiblePos = gZeroVec3f;
 
         if (actorId == ACTOR_OBJ_GRASS) {
             grassElemActor = va_arg(args, ObjGrassElement*);
-            pos = grassElemActor->pos;
+            GrassIdPos = grassElemActor->pos;
+            collectiblePos = GrassIdPos;
         } else if (actorId == ACTOR_EN_KUSA) {
             kusaActor = va_arg(args, EnKusa*);
-            pos = kusaActor->actor.home.pos;
+            GrassIdPos = kusaActor->actor.home.pos;
+            collectiblePos = kusaActor->actor.world.pos;
         } else if (actorId == ACTOR_OBJ_GRASS_CARRY) {
             grassCarryActor = va_arg(args, ObjGrassCarry*);
-            pos = grassCarryActor->actor.world.pos;
+            collectiblePos = grassCarryActor->actor.world.pos;
         }
 
         RandoCheckId randoCheckId =
-            IdentifyGrass(actorId == ACTOR_OBJ_GRASS_CARRY ? grassCarryActor->grassElem->pos : pos);
+            IdentifyGrass(actorId == ACTOR_OBJ_GRASS_CARRY ? grassCarryActor->grassElem->pos : GrassIdPos);
```

Comment:
Instead of this additional ternary for the grass carry pos, you can just set `GrassIdPos` above in the `else if`, like you do for the other two types.